dbms/src/Common/Arena.h

```diff
@@ -55,7 +57,9 @@ class Arena : private boost::noncopyable
             end = begin + size_ - pad_right;
             prev = prev_;
 
+#if __has_include(<sanitizer/asan_interface.h>)
```

Comment:
You can define macros `POISON_MEMORY_REGION` and `UNPOISON_MEMORY_REGION` in Defines.h that will be noop without ASan. It will make the code much more clean.